components/Buttons/src/TextButton/MDCButton+TextButton.m

```diff
@@ -0,0 +1,36 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCButton+TextButton.h"
+#import "MDCButtonColorThemer.h"
+#import "MDCButtonTypographyThemer.h"
+
+@implementation MDCButton (TextButton)
+
+- (instancetype)initTextButton {
+  self = [self init];
+  if (self) {
+    [MDCButtonColorThemer applySemanticColorScheme:[[MDCSemanticColorScheme alloc] init]
+                                      toFlatButton:self];
+    [MDCButtonTypographyThemer applyTypographyScheme:[[MDCTypographyScheme alloc] init]
+                                            toButton:self];
```

Comment:
If possible, we should try to avoid making our components depend on the scheme systems because we may change the system design in the future. Ideally the themer application is always done in client code.

Reply:
:/  We definitely was to avoid dependencies.  Is is acceptable that these live in Buttons+Extensions?